sdk/ml/azure-ai-ml/azure/ai/ml/operations/_batch_deployment_operations.py

```diff
@@ -106,7 +112,16 @@ def begin_create_or_update(
             operation_config=self._operation_config,
         )
         upload_dependencies(deployment, orchestrators)
-
+        if is_private_preview_enabled() and deployment.job_definition:
+            component = self._validate_component(deployment)
```

Comment:
Is this right? We validate and don't assign the `component` we validated?

Reply:
we just need to assign the component_id value